### PR TITLE
Development

### PR DESCRIPTION
**Pull request by cjjavellana**
# What
Merging development branch to prevent it from going too big to be merged later on.

Contains various WIP codes involving kafka and plugin configurations.

### CODE REVIEW

File: messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @Bean
+    open fun producerFactory(): ProducerFactory<String, String> {
+        return DefaultKafkaProducerFactory(producerConfigs())
+    }
+
+    @Bean
+    open fun producerConfigs(): Map<String, Any> {
+        // See https://kafka.apache.org/documentation/#producerconfigs for more properties
+        return mapOf(
+            ProducerConfig.BOOTSTRAP_SERVERS_CONFIG to "localhost:9092",
```
Comment (cuongnc01655) [inline messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt:39]: We can make this configurable

File: messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+}
+
+@Configuration
+open class KafkaConsumerConfig {
+    @Bean
+    open fun consumerFactory(): ConsumerFactory<String?, String?> {
+        return DefaultKafkaConsumerFactory(
+            mapOf(
+                ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG to "localhost:9092",
```
Comment (cuongnc01655) [inline messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt:57]: Here also

**Review by cuongnc01655 — COMMENTED** [review]
I think we can merge, just need to make Kafka endpoint/topic become configurable

File: messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt (hunk 1 above)
Comment (cjjavellana) [reply messaging-service/src/main/kotlin/com/forge/messageservice/configurations/KafkaConfiguration.kt]: Done done, removed custom configuration for producer & consumer. Kafka configuration is handled by `KafkaAutoConfiguration`

**Pull request merged**